File: TPoint/TPoint/Domain/Entities/Round.swift
//
//  Round.swift
//  TPoint
//

import Foundation

struct TichuCall: Hashable {
    let player: Player
    let isLarge: Bool  // true: 라지티츄(200), false: 스몰티츄(100)
    var isSuccess: Bool

    var points: Int {
        let basePoints = isLarge ? 200 : 100
        return isSuccess ? basePoints : -basePoints
    }

    var displayName: String {
        isLarge ? L10n.largeTichu : L10n.smallTichu
    }
}

struct Round {
    let roundNumber: Int
    var teamACardScore: Int
    var isOneTwoFinish: Bool
    var oneTwoFinishTeam: TeamType?
    var tichuCalls: [TichuCall]

    var teamBCardScore: Int {
        100 - teamACardScore
    }

    init(roundNumber: Int,
         teamACardScore: Int = 50,
         isOneTwoFinish: Bool = false,
         oneTwoFinishTeam: TeamType? = nil,
         tichuCalls: [TichuCall] = []) {
        self.roundNumber = roundNumber
        self.teamACardScore = teamACardScore
        self.isOneTwoFinish = isOneTwoFinish
        self.oneTwoFinishTeam = oneTwoFinishTeam
        self.tichuCalls = tichuCalls
    }
}

File: TichuTallyAndroid/app/src/main/java/com/tichutally/app/domain/model/Round.kt
package com.tichutally.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Round(
    val roundNumber: Int,
    val teamACardScore: Int = 50,
    val isOneTwoFinish: Boolean = false,
    val oneTwoFinishTeam: TeamType? = null,
    val tichuCalls: List<TichuCall> = emptyList()
) {
    val teamBCardScore: Int
        get() = 100 - teamACardScore
}
